__tests__/app.test.js

```diff
@@ -358,3 +358,33 @@ describe("GET /api/users", () => {
       });
   });
 });
+
+describe("200: GET /api/articles?topic=[input] should respond with all articles with a topic matching the query", () => {
+  test("200: GET /api/articles?topic=cats should respond with all articles with a topic of cats", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        console.log(body);
+        body.articles.forEach((article) => {
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("404: GET /api/articles?topic=[non-existent topic] should respond with Not Found if passed a topic that doesn't exist", () => {
+    return request(app)
+      .get("/api/articles?topic=pasta")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not Found");
+      });
+  });
+  test("404: GET /api/articles?topic=paper should respond with Not Found if query points to a topic that exists, but has no articles", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not Found");
+      });
+  });
```

Comment:
This should actually be a passing test, if the topic exists but no articles are associated with it then we should expect an empty array.